src/app/(mobile-ui)/home/page.tsx

```
@@ -107,11 +117,20 @@ export default function Home() {
         setFocusedIndex(targetIndex)
 
         if (targetIndex < wallets.length) {
-            dispatch(walletActions.setFocusedWallet(wallets[targetIndex]))
-
             const targetWallet = wallets[targetIndex]
-            if (targetWallet.walletProviderType === WalletProviderType.PEANUT || isWalletConnected(targetWallet)) {
-                setSelectedWallet(targetWallet)
+            dispatch(walletActions.setFocusedWallet(targetWallet))
+
+            // check wallet type and ID
+            const isValidPeanutWallet =
```

question: why is this necessary? In what cases do we have invalid wallets?

I prefer to fail and discover those cases than silently failing. Or is there something I'm not understanding?

---

@jjramirezn  two reasons why i added these checks:
- to select/focus a wallet card, i was having issues with selecting/focusing the rewards card, as it shared same address as peanut wallet, but id based selection solved it
- second, to ensure proper carousel behavior, without these valid check, it would not let me scroll/click through disconnected wallets and go to my rewards wallet

---

open to any other suggestion you might have or a better solution tbh, even i'm not really sure if this a good fix, but it worked :*)

---

Agree, the way to go would be to see what about the wallet selector or wallet hook is causing that. I'm approving and taking a note, I have a task for the wallet selector assigned so I'll take a look when I'm around there